lib/screens/Badge/index.dart

```diff
@@ -128,33 +140,68 @@ class _BadgePageState extends State<BadgePage> {
 
   dynamic sendRequest(String userId, int badgeId) async {
     // get auth token
-      SharedPreferences prefs = await SharedPreferences.getInstance();
-      String token = prefs.getString('token');
-
-      // set request params
-      var body = {
-        "redeem": {"attendee_id": userId, "badge_id": badgeId.toString()}
-      };
-      String jsonBody = json.encode(body);
-      print(jsonBody);
-      final encoding = Encoding.getByName('utf-8');
-
-      // send request
-      return await http.post(
-          DotEnv().env['API_URL'] + 'api/v1/redeems',
-          headers: {
-            "Content-Type": "application/json",
-            "Authorization": "Bearer " + token
-          },
-          body: jsonBody,
-          encoding: encoding);
+    SharedPreferences prefs = await SharedPreferences.getInstance();
+    String token = prefs.getString('token');
+
+    // set request params
+    var body = {
+      "redeem": {"attendee_id": userId, "badge_id": badgeId.toString()}
+    };
+    String jsonBody = json.encode(body);
+    print(jsonBody);
+    final encoding = Encoding.getByName('utf-8');
+
+    // send request
+    return await http.post(DotEnv().env['API_URL'] + 'api/v1/redeems',
+        headers: {
+          "Content-Type": "application/json",
+          "Authorization": "Bearer " + token
+        },
+        body: jsonBody,
+        encoding: encoding);
+  }
+
+  scanQRCodeContinuously(int badgeId) {
+    FlutterBarcodeScanner.getBarcodeStreamReceiver(
+            "#006064", "Cancel", true, ScanMode.QR)
+        .listen((link) async {
+      RegExp regExp = new RegExp(
+        ".*https:\/\/enei.pt\/user\/(([A-Za-z0-9]+-*)+)",
+        caseSensitive: false,
+        multiLine: false,
+      );
+
+      // check if link is valid
+      if (regExp.hasMatch(link)) {
+        List<String> vars = link.split("/");
+        String userId = vars[vars.length - 1];
+
+        final response = await sendRequest(userId, badgeId);
+
+        // change screen state regarding request result
+        if (response.statusCode == 201) {
```

Comment:
please change to:

 // change screen state regarding request result
        if (response.statusCode == 201) {
          if (await Vibration.hasVibrator()) {
            // long vibrate : correct
            Vibration.vibrate(duration: 1000);
          }
        } else {
          if (await Vibration.hasVibrator()) {
            // double vibrate : error
            Vibration.vibrate(pattern: [0, 500, 250, 500]);
          }
        }
      } else {
        if (await Vibration.hasVibrator()) {
          // double vibrate : error
            Vibration.vibrate(pattern: [0, 500, 250, 500]);
        }